pandas/core/frame.py

```diff
@@ -6741,16 +6742,38 @@ def _gotitem(
         **_shared_doc_kwargs,
     )
     @Appender(_shared_docs["aggregate"])
-    def aggregate(self, func, axis=0, *args, **kwargs):
+    def aggregate(self, func=None, axis=0, *args, **kwargs):
```

Comment:
Why did this need to change?

Reply:
this is needed to make it optional, so that e.g. `df.agg(bar=('a', max))` could be handled correctly